js/mviewerstudio.js

```diff
@@ -1,838 +1,1098 @@
-var _conf;
-var API = {};
-var VERSION = "3.1-snapshot";
+const jquery = require('jquery')
+$ = window.$ = window.jQuery = jquery;
+import mv from "../lib/mv";
+import ogc from "../lib/ogc";
+//import slug from "../lib/slug"
 
-$(document).ready(function(){
-   
-    //Mviewer Studio version
-    console.log("MviewerStudio version " + VERSION);
-    
-    //Get URL Parameters
-    if (window.location.search) {
-        $.extend(API, $.parseJSON('{"' + decodeURIComponent(window.location.search.substring(1)
-            .replace(/&/g, "\",\"").replace(/=/g,"\":\"")) + '"}'));
-    }
-    $.ajax({
-        type: "GET",
-        url: "config.json",
-        dataType: "json",
-        contentType: "application/json",
-        success: function (data) {
-            console.groupCollapsed("init app from config");
-            _conf = data.app_conf;
-            if (_conf.proxy === undefined) {
-                _conf.proxy = "../proxy/?url=";
-            }
+// OpenLayers
+import Map from "ol/Map";
```

Comment:
Perso, je préfèrerais qu'on garde la mention ol. avant chaque utilisation de cette bibliothèque.
Je sais pas c'est peut-être juste une habitude ou la peur de tomber sur une autre bibliothèque qui proposerait une classe, variable dont le nom est déjà utilisé par OpenLayers.

Reply:
On peut faire ça `import {Map as olMap} from 'ol'` mais si on fait `import 'ol'` alors on perd le bénéfice du tree-shaking et on va inclure l’entièreté d'openlayers dans la lib alors qu'on utilise qu'une tout petite partie. Pour reduire la confusion, il faudrait surtout splitter l'appli en plusieurs fichiers (par exemple un fichier `map.js`) plus petit qui rendent la confusion pratiquement impossible. Mais le but ici c'etait de montrer qu'en gros avec 1h30/2h de taff on ajoute un bundler et après on peut itérer dessus.

Reply:
Préfixer ce que l'on importe par ol me paraît une bonne idée mais il faudrait confronter cela aux pratiques sur d'autres projets. Ce serait bien qu'on ait des pratiques similaires.
Pour ce qui est de splitter l'application : c'est une bonne idée. Il faut absolument qu'on le fasse et ton initiative me paraît être le bon déclencheur. L'intérêt c'est pas tellement d'éviter les confusions pour des noms identiques entre bibliothèques mais plutôt d'éviter d'avoir des fichiers de plusieurs milliers de lignes. Un code mieux structuré nous permettra sans doute de plus facilement le maintenir et le faire évoluer.